features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/RumSessionEndedMetricTest.kt

```diff
@@ -0,0 +1,412 @@
+            session = event1.session.copy(id = fakeSessionId),
+            date = date1,
+            view = event1.view.copy(timeSpent = duration1)
+        )
+        val differentSessionViewEvent = event2.copy(
+            date = date2,
+            view = event2.view.copy(timeSpent = duration2)
+        )
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+
+        // When
+        sessionEndedMetric.onViewTracked(currentSessionViewEvent)
+        sessionEndedMetric.onViewTracked(differentSessionViewEvent)
+
+        // Then
+        val attributes = sessionEndedMetric.toMetricAttributes()
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val viewCounts = rse[SessionEndedMetric.VIEW_COUNTS_KEY] as Map<*, *>
+        assertEquals(1, viewCounts[SessionEndedMetric.VIEW_COUNTS_TOTAL_KEY])
+    }
+
+    @Test
+    fun `M view counts total correct W track views`(@Forgery events: List<ViewEvent>) {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+
+        // When
+        events.map {
+            it.copy(session = it.session.copy(id = fakeSessionId))
+        }.forEach {
+            sessionEndedMetric.onViewTracked(it)
+        }
+        // Then
+        val attributes = sessionEndedMetric.toMetricAttributes()
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val viewCounts = rse[SessionEndedMetric.VIEW_COUNTS_KEY] as Map<*, *>
+        assertEquals(events.count(), viewCounts[SessionEndedMetric.VIEW_COUNTS_TOTAL_KEY])
+    }
+
+    @Test
+    fun `M view counts background correct W track views`(@Forgery events: List<ViewEvent>) {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+        val backgroundViewCount = Random.nextInt(events.size)
+
+        // When
+        events.mapIndexed { index, viewEvent ->
+            val url = if (index in 0 until backgroundViewCount) {
+                RumViewManagerScope.RUM_BACKGROUND_VIEW_URL
+            } else {
+                viewEvent.view.url
+            }
+            viewEvent.copy(
+                session = viewEvent.session.copy(id = fakeSessionId),
+                view = viewEvent.view.copy(url = url)
+            )
+        }.forEach {
+            sessionEndedMetric.onViewTracked(it)
+        }
+        // Then
+        val attributes = sessionEndedMetric.toMetricAttributes()
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val viewCounts = rse[SessionEndedMetric.VIEW_COUNTS_KEY] as Map<*, *>
+        assertEquals(backgroundViewCount, viewCounts[SessionEndedMetric.VIEW_COUNTS_BG_KEY])
+    }
+
+    @Test
+    fun `M view counts app launch correct W track views`(@Forgery events: List<ViewEvent>) {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+        val appLaunchViewCount = Random.nextInt(10)
```

Comment:
bonus point: in case of failure you will get a seed printed, which can be used to reproduce the failure.

```suggestion
    fun `M view counts app launch correct W track views`(
      @Forgery events: List<ViewEvent>,
      @IntForgery(min = 1, max = 10) appLaunchViewCount
    ) {
        // Given
        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
```

Reply:
Good point, updated